src/commcare_cloud/ansible/roles/zookeeper/templates/zoo.cfg.j2

```diff
@@ -0,0 +1,54 @@
+# http://hadoop.apache.org/zookeeper/docs/current/zookeeperAdmin.html
+
+# The number of milliseconds of each tick
+tickTime=2000
+# The number of ticks that the initial
+# synchronization phase can take
+initLimit=10
+# The number of ticks that can pass between
+# sending a request and getting an acknowledgement
+syncLimit=5
+# the directory where the snapshot is stored.
+dataDir={{ zookeeper_data_dir }}
+# Place the dataLogDir to a separate physical disc for better performance
+# dataLogDir=/disk2/zookeeper
+
+# the port at which the clients will connect
+clientPort=2181
+# specify all zookeeper servers
+# The fist port is used by followers to connect to the leader
+# The second one is used for leader election
+
+# The Current node IP should be replaced with 0.0.0.0 in each node.
+{% for host in groups['zookeeper'] %}
+{% if host == inventory_hostname %}
+server.{{ host }} = 0.0.0.0:2888:3888
+{% else %}
+server.{{ host }} = {{ host }}:2888:3888
+{% endif %}
+{% endfor %}
+
+# To avoid seeks ZooKeeper allocates space in the transaction log file in
+# blocks of preAllocSize kilobytes. The default block size is 64M. One reason
+# for changing the size of the blocks is to reduce the block size if snapshots
+# are taken more often. (Also, see snapCount).
+#preAllocSize=65536
+# Clients can submit requests faster than ZooKeeper can process them,
+# especially if there are a lot of clients. To prevent ZooKeeper from running
+# out of memory due to queued requests, ZooKeeper will throttle clients so that
+# there is no more than globalOutstandingLimit outstanding requests in the
+# system. The default limit is 1,000.ZooKeeper logs transactions to a
+# transaction log. After snapCount transactions are written to a log file a
+# snapshot is started and a new transaction log file is started. The default
+# snapCount is 10,000.
+#snapCount=1000
+
+# If this option is defined, requests will be will logged to a trace file named
+# traceFile.year.month.day.
+#traceFile=
+
+# Leader accepts client connections. Default value is "yes". The leader machine
+# coordinates updates. For higher update throughput at thes slight expense of
+# read throughput the leader can be configured to not accept clients and focus
+# on coordination.
+#leaderServes=yes
```

Comment:
Can you explain where this file came from? Was it copied from another place in our codebase? Or from the default config file that comes with the zookeeper install?

Reply:
I have copied it from the kafka2 server.

Reply:
Okay so it sounds like is the default that comes with the zookeeper install. I'm a little surprised that we have to make this change because I think a few years ago we already a multi node zookeeper setup, but I think it's fine.

Could you post on this ticket when you apply it what the actual change looks like (the ansible diff)? It's just the lines like
```
server.{{ host }} = {{ host }}:2888:3888
```
that are different, right?

Reply:
Sure, Danny.